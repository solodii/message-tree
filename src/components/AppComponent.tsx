import * as React from 'react';
import MessageTreeContainer from '../containers/MessageTreeContainer';

export interface Props {
  initApp: () => void;
}

export default class AppComponent
  extends React.Component<Props, {}>
  implements React.ComponentLifecycle<Props, {}> {

  public componentDidMount(): void {
    this.props.initApp();
  }

  public render(): JSX.Element {
    return <MessageTreeContainer />;
  }
}
