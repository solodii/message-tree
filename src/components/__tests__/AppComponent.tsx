import * as React from 'react';
import { shallow } from 'enzyme';
import AppComponent, { Props } from '../AppComponent';

describe('AppComponent', () => {
  const initApp = jest.fn();
  const initialProps: Props = { initApp };
  const wrapper = shallow<Props, {}>(
    <AppComponent {...initialProps} />
  );

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('initApp is called after componentDidMount', () => {
    expect(initApp).toHaveBeenCalledTimes(0);
    (wrapper.instance() as AppComponent).componentDidMount();
    expect(initApp).toHaveBeenCalledTimes(1);
  });
});
