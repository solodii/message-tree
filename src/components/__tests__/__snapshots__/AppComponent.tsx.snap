// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent renders correctly 1`] = `<MessageTreeContainer />`;
